### DISCUSSION TITLE
Installazione HA su MiniPC

### DISCUSSION THREAD

**Discussion by lucaxxaa** [indomus/forum#2865] [HUB: Home Assistant]
Argomento gia' trattato odi cui ho letto i post precedenti, ed anche [le guida sul sito](https://indomus.it/formazione/home-assistant-su-mini-pc-intel-nuc-le-varie-installazioni-le-cose-da-fare/), ma mi rimangono dei dubbi.
Premessa, sono un principiante, uso HA OS su Rpi4 da 2 anni. 
Mi hanno regalato un miniPC con intel N100 , 16 GB Ram ddr5, e vorrei usare questo, ma **non sacrificarlo solo per HA OS,** ma usarlo anche per altro, e **non sacrificarlo  usando proxmox**.
Leggendo le varie guide sarei arrivato alla conclusione, non consigliata sul sito, di installare debian su minipc, installare docker, e HA OS su docker, in quanto vorrei poter aggiornare HA OS semplicemente e fare i backup, come faccio ora su RPI4, senza preoccuparmi di altro. 
Ma con tale installazione :
1-il mio attuale backup di HA OS su Rpi4 posso trasferirlo nella futura installazzione ?
2-ma posso aggiornare il sistema host debian regolarmente e nel tempo, o rischio che qualche dipendenza di HA OS su docker ne comprometta l'utilizzo ? ( per tale punto come comportarsi nel tempo, per ovviare a problemi) 
3-il backup automatici di HA OS, funzionano ho letto, esatto?

Ovviamente accetto consigli.

**Comment by snakuzzo** [comment]
Il problema è il punto 2. Con quel tipo di installazione, che noi sconsigliamo vivamente, se aggiorni anche solo un pacchetto che non devi aggiornare...e ti incasinerai.

Se posso darti un consiglio...nel tuo caso sarebbe meglio installare il sistema operativo che ti pare e tirare su una VM con HA OS.

**Reply by lucaxxaa** [reply]
Grazie. Al momento sono in fase di valutazione.

**Reply by crug80** [reply]
Perché usare proxmox sarebbe sacrificarlo?

**Reply by KC-inDomus** [reply]
ecco perché: https://indomus.it/guide/home-assistant-hassio-o-home-assistant-core-quale-installazione-fa-al-caso-mio/

**Reply by crug80** [reply]
Secondo me il problema principale non è tanto proxmox si o no, ma la volontà di usare hardware minimal per più cose. La virtualizzazione ha rivoluzionato il mondo server, aggiungendo sì un layer ulteriore di complessità, ma facilitando in modo enorme la gestione, i backup, le procedure di disaster recovery e la scalabilità. Tutto questo ha avuto un costo in termini di requisiti hardware (cpu + ram). In base alla mia esperienza sintetizzerei la cosa in: proxmox/esxi/hyper visor con HAOS sì, a patto di usare un hardware decente, altrimenti non ha senso. Un mini pc non può essere un NAS, un server di autenticazione ed un hub di domotica, non è stato pensato per quello, IMHO.

**Reply by trebestie** [reply]
Ho un Fujitsu Futro s920 con 4gb di RAM. Ci faccio girare da oltre un anno una decina di container LXC tra cui uno con ubuntu e home assistant core.
Se non ci devi giocare o fare rendering 3d non capisco quali problemi possano esserci su una macchina con 16GB di RAM

**Reply by lucaxxaa** [reply]
Infatti ho un Intel n100 basso consumo, con DDR5, proprio per tenerlo acceso H24 e consumare pochi watt.
Sarà per uso esclusivo, di haos, che solo per questo sarebbe altamente sprecato perché basta un Raspberry anche il 3,  e appunto mi stavo guardando intorno, per poter usare il Mini PC anche per altri usi ad esempio come NAS casalingo.
E sono approdato a trunas che mi sembra una buona soluzione.

**Comment by lucaxxaa** [comment]
Stavo considerando openmediavalut come distribuzione principale, che si basa su Debian.
Con questa il minipc potrei usarlo per altro, e usare ha, in docker.

C'è qualche guida facile su come fare backup ha core, così in caso di problemi si ripristinerà facilmente

Posso, usare il mio attuale backup di Ha Os, sulla nuova installazione di ha composer ?

**Reply by snakuzzo** [reply]
Non ho capito che vuoi fare.
Parli di ha os, poi di "ha composer" che non so che diavolo sarebbe.
Che vuoi fare? Che installazione vuoi?
HA Supervised su docker o core su docker?

Ha supervised, come già detto, la sconsigliamo a prescindere...e a maggior ragione se non utilizzi Debian (OMV è basata su Debian, ma non è Debian).

Se invece, partendo da ha os, vuoi installare core su docker ripristinando un backup, puoi solo recuperare dallo zip la configurazione di core e utilizzare quella...ma quasi sicuramente dovrai fare qualche adattamento.

**Comment by lucaxxaa** [comment]
Sto valutando.
Se usassi openmediavalut ( che e basata su Debian) potri installare

1-' ha core e tutti gli altri componenti ( esphome, ec....) su docker.
In questo caso, cercavo se esiste qualche guida su come fare backup periodicamente e come ripristinare.
E se posso usare il mio attuale backup di haos, per ripristinare database ha

2-installare ha supervised su docker sempre sotto host openmediavalut, ma penso che valga stesso discorso del secondo post

**Reply by snakuzzo** [reply]
per favore, la finisci di creare nuovi thread sotto la stessa discussione ? Fai reply sotto l'ultimo messaggio. Grazie.

1. Core su Docker lo puoi installare su qualsiasi distribuzione Windows, Linux, macOS o altro. 
Per il backup di core ci sono innumerevoli soluzioni. Comincia a dare un'occhiata a questo -> https://indomus.it/guide/effettuare-il-backup-ed-eventuale-ripristino-di-home-assistant-core-su-raspbian , ma, come ti dicevo, non è l'unico modo.
Ti ho già detto che il tuo attuale backup di HA OS puoi usarlo solo in parte. Apri lo zip e recuperi la config dir di Core...e usi quella quando andrai ad istanziare il container...però quasi sicuramente dovrai apportare qualche modifica (lo scoprirai direttamente quando lo farai partire guardando i log).

2. Lo ripeto per l'ennesima volta, poi basta. HA SUPERVISED LO SCONSIGLIAMO A PRESCINDERE! Inoltre, NON E' SUPPORTATO SU OMV !!! E' supportato solo su Debian Stable (PULITA) e sarai sempre vincolato a dover essere su ultima release, oltre a dover tenere determinati pacchetti sempre aggiornati a specifiche release indicate nei prerequisiti. Se per caso aggiorni anche solo un pacchetto che non devi aggiornare, l'installazione che hai ti risulterà non supportata. Ricorda che è un'installazione consigliata solo a persona con un certo background e che su linux si sanno muovere senza problemi...e non mi pare il tuo caso a questo punto.

**Reply by lucaxxaa** [reply]
Grazie.
Ho letto l articolo su backup ha core. Non sembra difficile.
Avrei altre domande, se fossi così gentile a rispondere
1- in caso ha core sia installato su docker, prima di eseguire i comandi di backup della guida, bisogna fermare il docker?( c'è qualche guida?)
2- in docker oltre ad ha core, installerei anche altri componenti, tra cui esphome, nuc, duckdns, fileditor, ecc) . Dovrei fare la copia di backup di ogni container, esatto?
Grazie e scusa l ignoranza

**Reply by snakuzzo** [reply]
1. Non c'è bisogno di fermare Core...e non c'è bisogno di alcuna guida per capire come fermare core, nel caso trovi comunque tutto nelle guide di installazione. Il mio consiglio però è quello di studiare prima come funziona Docker, non puoi andare così alla cieca senza nemmeno sapere queste cose.
2. Puoi provare a recuperare le configurazioni degli addon e riportarle nella definizione dei container, ma con un semplice copia-incolla di sicuro no.

**Reply by lucaxxaa** [reply]
Premesso che non voglio fare polemiche, ed il titolo del post ed il mio intento e' di lasciare aperte ad una moltitudine di strade diverse, e che sono in fase di sperimentazione, ho visto e sto provando come test, su macchina virtuale, True Nas scale.
Con True Nas, che mi userei esempio come principale sistema installato su Host, si puo:

-creare macchina virtuale sempre a mezzo di Truenas, dove installare HAOS, che e' la cosa che voglio provare appena mi arriva il mini pc, e sarebbe quella definitiva per me, usando truenas, haos in modo semplice, ed eventuali altre app di truenas
-installare ha core su container( app gia semp0lificata di truenas) , che ho gia' fatto

per ora sto solo sperimentando, e finche non trovo una soluzione definitiva, tengo Haos su rpi4.

**Reply by 50494554524F** [reply]
sto provando haos sia su rpi che n100 come macchina virtuale con debian gnome + cockpit + virt-manager (per le funzioni che ancora mancano in cockpit)... ormai si può fare tutto da interfaccia; rpi e n100 hanno la virtualizzazione avanzata quindi le vm girano bene e si può mantenere haos senza sbattersi... devo ancora provarle in situazione reale, ma penso che andrà sicuramente bene... n100 è sicuramente sprecato se non si usano videocamere...

**Reply by 50494554524F** [reply]
ti aggiorno sulle prove che ho effettuato... raspberry 4, più lento ed i tempi si vedono meglio

microsd (boot) + hdd (data) + haos 2024.5
hdd (boot+data) + haos 2024.5
hdd (boot+data) + ubuntu rpi 24.04 avvio senza gui + haos 2024.5 in vm

esattamente gli stessi di tempi di avvio/riavvio praticamente al secondo; la mia configurazione fa avvio host in 7:30m e riavvia core in 3:30m (escluso il caricamento delle dashboard, che richiede altro tempo per averle disponibili)... considerando i tempi di avvio di ubuntu, 1:30m da spento, l'installazione più veloce sarebbe haos in vm

considerando che con ubuntu ora c'è un remote desktop che funziona come si deve, posso evitarmi di usare il terminal, terrò questa con un minimo di limitazioni date dalle vm

mi manca solo da misurare i consumi in situazione reale...

**Reply by KC-inDomus** [reply]
questo thread è surreale
